### ISSUE TITLE
About the input of loss_contrastive

### ISSUE THREAD

**Issue by dwbzxc** [PRBonn/ContMAV#1]
Thanks for your great work! I have a question when reading the source code: Should the features input of the loss_contrastive be pred_scales from the Semantic Decoder or ow_res from the Contrastive Decoder? (See the training code: https://github.com/PRBonn/ContMAV/blame/715238f82c421250bb66e6d1b1a85ded776e5abf/train.py#L284. This is different from the input in the validation code: https://github.com/PRBonn/ContMAV/blame/715238f82c421250bb66e6d1b1a85ded776e5abf/train.py#L427)

Thank you!

**Comment by matteosodano** [comment]
Hello,
sorry for the confusion. It's `ow_res`. Fixed it. Thank you for noticing!

**Comment by dwbzxc** [comment]
Got it. Thanks for your answer!

**Comment by ChenJiaJiaJia** [comment]
> Hello, sorry for the confusion. It's `ow_res`. Fixed it. Thank you for noticing!

the latest branch doesn't fix it .  It seems like something wrong about merging branchs

**Comment by matteosodano** [comment]
I think it does.

Line 262: `pred_scales, ow_res = model(image)`
Line 276: `loss_objectosphere = loss_obj(ow_res, label_ow)`
Line 279: `loss_ows = loss_mav(pred_scales, label, is_train=True)`
Line 282: `loss_con = loss_contrastive(mavs, ow_res, label, epoch)`

Objectosphere and contrastive loss are active on the second decoder, so they use `ow_res`, while the feature loss (`mav_loss`) acts on the first one, so it uses `pred_scales`.

**Comment by ChenJiaJiaJia** [comment]
您的邮件我已经收到，我会第一时间给您回复，谢谢！                                     陈佳佳

**Comment by ChenJiaJiaJia** [comment]
oh ,sorry, I made a mistake. the code I saw is from previous commit.
